src/overlays/actors/ovl_Obj_Bell/z_obj_bell.c

```diff
@@ -21,26 +29,259 @@ const ActorInit Obj_Bell_InitVars = {
+            break;
+    }
+
+    this->unk_21C = CLAMP(this->unk_21C, 0.0f, 18000.0f);
+    if (phi_a3 == 1) {
+        Math_Vec3s_ToVec3f(&bumperPos, &this->collider2.info.bumper.hitPos);
+        Math_Vec3f_Copy(&newVec, &this->dyna.actor.world.pos);
+        this->dyna.actor.world.rot.y = Math_Vec3f_Yaw(&bumperPos, &newVec);
+        if (this->unk_20C <= 0x4000 && this->unk_20C >= -0x4000) {
+            this->unk_20C -= 0x4000;
+        } else {
+            this->unk_20C += 0x4000;
+        }
+    }
+    return phi_a3;
+}
+
+s32 func_80A356D8(ObjBell* this) {
+    f32 temp_f2;
+    s16 new_var;
+
+    this->unk_212 = this->dyna.actor.world.rot.y;
+    if (this->unk_20C >= 0x4000 || this->unk_20C < -0x3FFF) {
+        this->unk_212 -= 0x8000;
+    }
+    temp_f2 = Math_SinS(this->unk_20C) * this->unk_21C;
+    this->dyna.actor.world.rot.x = this->dyna.actor.home.rot.x;
+    new_var = temp_f2;
+    this->unk_220 = temp_f2;
+    this->dyna.actor.world.rot.x += new_var;
+    Math_ApproachF(&this->unk_21C, 0.0f, 0.03f, 70.0f);
+    if (this->unk_21C > 0.0f) {
+        this->unk_20C -= 0x800;
+    }
+    return 0;
+}
+
+s32 func_80A357A8(ObjBell* this, GlobalContext* globalCtx) {
+    f32 temp_f0;
+    s16 temp_v1;
+
+    if ((this->collider1.base.ocFlags1 & 2)) {
+        this->collider1.base.ocFlags1 &= 0xFFFD;
+        temp_v1 = this->dyna.actor.yawTowardsPlayer - this->unk_212;
+        temp_f0 = this->unk_21C / 18000.0f;
+        if (ABS_ALT(temp_v1) < 0x3FFC) {
+            if (this->unk_214 == 0) {
+                if (temp_f0 > 0.18f) {
+                    func_800B8D98(globalCtx, &this->dyna.actor, 8.0f * temp_f0, this->dyna.actor.yawTowardsPlayer,
+                                  11.0f * temp_f0);
+                    this->unk_214 = 30;
+                }
+            }
+        }
+    }
+    if ((this->collider2.base.acFlags & 2)) {
+        this->collider2.base.acFlags &= 0xFFFD;
+        this->unk_20E = 10;
+        switch (this->dyna.actor.colChkInfo.damageEffect) {
+            case 15:
+                Audio_PlayActorSound2(this, 0x28F3);
+                func_80A35510(this, 1);
+                break;
+            case 14:
+                Audio_PlayActorSound2(this, 0x28F3);
+                func_80A35510(this, 2);
+                break;
+            default:
+                func_80A35510(this, 0);
+                break;
+        }
+    }
+    return 0;
+}
+
+void func_80A358FC(ObjBell* this, GlobalContext* globalCtx) {
+    s16 phi_v1;
+
+    this->collider1.dim.worldSphere.radius = (this->collider1.dim.modelSphere.radius * this->collider1.dim.scale);
+    this->collider2.dim.worldSphere.radius = (this->collider2.dim.modelSphere.radius * this->collider2.dim.scale);
+
+    // TODO: This temp could be taken out, but do we want it in for readability?...
+    phi_v1 = DECR(this->unk_20E);
+    if (phi_v1 == 0) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider2.base);
+    }
```

Comment:
```suggestion
    if (DECR(this->unk_20E) == 0) {
        CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider2.base);
    }
```
Yeah, I'd take this temp out:
https://discord.com/channels/688807550715560050/688851997352263820/844406587870937119